### PR TITLE
Ported to use reusable sequence parser for uint

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**

### CODE REVIEW

File: src/Channels/ReadableBuffer.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public struct ReadableBuffer : ISequence<ReadOnlyMemory<byte>>
         /// </summary>
         public int Length => _length >= 0 ? _length : GetLength();
 
-        int? ISequence<ReadOnlyMemory<byte>>.Length => Length;
+        int? ISequence<ReadOnlyMemory<byte>>.Length => _length >= 0 ? _length : (int?)null;
```
Comment (davidfowl) [inline src/Channels/ReadableBuffer.cs]: Shouldn't it compute the length?

File: test/Channels.Tests/project.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "Channels.Networking.Windows.RIO": {
       "target": "project"
     },
-    "xunit": "2.2.0-beta2-build3300"
+    "xunit": "2.2.0-beta2-build3300",
+    "System.Collections.Sequences": "0.1.0-e161104-6"
```
Comment (davidfowl) [inline test/Channels.Tests/project.json]: Use 0.1.0-*

File: src/Channels.Text.Primitives/ReadableBufferExtensions.cs [hunk 3]
```diff
@@ -78,35 +79,12 @@ private static bool IsWhitespaceChar(int ch)
...
-                textSpan = new ReadOnlySpan<byte>(buffer.ToArray());
-            }
-
             uint value;
-            var utf8Buffer = new Utf8String(textSpan);
-            if (!PrimitiveParser.TryParseUInt32(utf8Buffer, out value))
-            {
-                throw new InvalidOperationException();
+            int consumed;
+            if(!buffer.TryParseUInt32(out value, out consumed)){
```
Comment (davidfowl) [inline src/Channels.Text.Primitives/ReadableBufferExtensions.cs]: nit: Fix the brace formatting 😄

**Review by davidfowl — CHANGES_REQUESTED** [review]

File: src/Channels/ReadableBuffer.cs (hunk 1 above)
Comment (KrzysztofCwalina) [reply src/Channels/ReadableBuffer.cs]: the point of int? is that if the collection has the length it returns it; if it does not, then the consumer can compute it.

File: test/Channels.Tests/project.json (hunk 2 above)
Comment (KrzysztofCwalina) [reply test/Channels.Tests/project.json]: fixed

File: src/Channels.Text.Primitives/ReadableBufferExtensions.cs (hunk 3 above)
Comment (KrzysztofCwalina) [reply src/Channels.Text.Primitives/ReadableBufferExtensions.cs]: fixed

File: src/Channels/ReadableBuffer.cs (hunk 1 above)
Comment (davidfowl) [reply src/Channels/ReadableBuffer.cs]: So lazy length = not having a length? It's just lazy....

Comment (KrzysztofCwalina) [reply src/Channels/ReadableBuffer.cs]: Ah, makes sense. I missed that.

**Review by davidfowl — APPROVED** [review]

**Review by davidfowl — APPROVED** [review]

**Pull request merged**